app/commands/google_service.py

```diff
@@ -25,25 +27,10 @@ def open_modal(client, body):
     client.views_open(trigger_id=body["trigger_id"], view=view)
 
 
-def google_service_command(client, body):
-    open_modal(client, body)
-
-
-def list_folders():
-    service = get_google_service("drive", "v3")
-    results = (
-        service.files()
-        .list(
-            pageSize=25,
-            supportsAllDrives=True,
-            includeItemsFromAllDrives=True,
-            corpora="drive",
-            q="parents in '{}' and mimeType = 'application/vnd.google-apps.folder' and trashed=false and not name contains '{}'".format(
-                SRE_INCIDENT_FOLDER, "Templates"
-            ),
-            driveId=SRE_DRIVE_ID,
-            fields="nextPageToken, files(id, name)",
-        )
-        .execute()
-    )
-    return results.get("files", [])
+def google_service_command(client, body, respond):
+    respond(f"Healthcheck status: {google_drive.healthcheck()}")
+    folders = google_drive.list_folders_in_folder(SRE_INCIDENT_FOLDER)
+    if not folders:
+        respond("The folder ID is invalid. Please check the environment variables.")
```

Comment:
Would that message be only if the folder ID is invalid or are there other cases - ie it can't communicate with google drive?

Reply:
Could you also log an error here just so we get pinged via the CloudWatch alarm.

Reply:
This is just a dev command to test the behaviour of certain functions. In this case, I was actually testing how the bot's service account permissions in Google Drive worked.

I modified the list_folders function to use the corpora="user" instead of "drive". Using drive required that the bot had at least read access to the entire share drive, whereas by using "user", we can grant the bot only access to the specific folders and documents to perform its job.